src/app/todo-card/todo-card.component.ts

```diff
@@ -0,0 +1,62 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { TodoModel, priorityType } from '../models/todo-model';
+import { AppComponent } from '../app.component';
+
+@Component({
+  selector: 'app-todo-card',
+  templateUrl: './todo-card.component.html',
+  styleUrls: ['./todo-card.component.scss']
+})
+export class TodoCardComponent implements OnInit {
+  @Input() todo: TodoModel;
+  @Input() appComponent: AppComponent;
+
+  constructor() { }
+
+  ngOnInit() {
+  }
+  
+  toggleComplete() {
+    this.appComponent.todoService.toggleCompleted(this.todo);
+    this.appComponent.updateTodoList();
+  }
+
+  deleteTodo() {
```

Comment:
You can possibly look into [EventEmitters](https://angular.io/api/core/EventEmitter) to help you pass a delete event to the parent component? Not sure if it works, but it'd be nice